pkg/cloud/services/ssm/secret_test.go

```diff
@@ -0,0 +1,293 @@
+			ms.SetSecretPrefix(tt.secretPrefix)
+			data := generateBytes(tt.bytesCount)
+
+			prefix, _, err := s.Create(ms, data)
+			check(prefix, tt.expectedPrefix, err, tt.wantErr)
+		})
+	}
+}
+
+func TestService_Delete(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	tests := []struct {
+		name        string
+		secretCount int32
+		expect      func(m *mock_ssmiface.MockSSMAPIMockRecorder)
+		wantErr     bool
+		check       func(error)
+	}{
+		{
+			name:        "Should not call AWS when secret count has zero value",
+			secretCount: 0,
+			expect:      func(m *mock_ssmiface.MockSSMAPIMockRecorder) {},
+		},
+		{
+			name:        "Should not return error when delete is successful",
+			secretCount: 1,
+			expect: func(m *mock_ssmiface.MockSSMAPIMockRecorder) {
+				m.DeleteParameter(gomock.Eq(&ssm.DeleteParameterInput{
+					Name: aws.String("prefix/0"),
+				})).Return(&ssm.DeleteParameterOutput{}, nil)
+			},
+		},
+		{
+			name:        "Should return all errors except not found errors",
+			secretCount: 3,
+			expect: func(m *mock_ssmiface.MockSSMAPIMockRecorder) {
+				m.DeleteParameter(gomock.Eq(&ssm.DeleteParameterInput{
+					Name: aws.String("prefix/0"),
+				})).Return(nil, awserrors.NewFailedDependency("failed dependency"))
+				m.DeleteParameter(gomock.Eq(&ssm.DeleteParameterInput{
+					Name: aws.String("prefix/1"),
+				})).Return(nil, awserrors.NewNotFound("not found"))
+				m.DeleteParameter(gomock.Eq(&ssm.DeleteParameterInput{
+					Name: aws.String("prefix/2"),
+				})).Return(nil, awserrors.NewConflict("new conflict"))
+			},
+			wantErr: true,
+			check: func(err error) {
+				if err.Error() != "[failed dependency, new conflict]" {
+					t.Fatalf("Unexpected error: %v", err)
+				}
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+			scheme := runtime.NewScheme()
+			_ = infrav1.AddToScheme(scheme)
+			client := fake.NewClientBuilder().WithScheme(scheme).Build()
+
+			clusterScope, err := getClusterScope(client)
+			g.Expect(err).NotTo(HaveOccurred())
+
+			ssmClientMock := mock_ssmiface.NewMockSSMAPI(mockCtrl)
+			tt.expect(ssmClientMock.EXPECT())
+			s := NewService(clusterScope)
+			s.SSMClient = ssmClientMock
+			ms, err := getMachineScope(client, clusterScope)
+			g.Expect(err).NotTo(HaveOccurred())
+
+			ms.SetSecretPrefix("prefix")
+			ms.SetSecretCount(tt.secretCount)
+
+			err = s.Delete(ms)
+			if tt.wantErr {
+				g.Expect(err).To(HaveOccurred())
+				if tt.check != nil {
+					tt.check(err)
+				}
+				return
+			}
+			g.Expect(err).NotTo(HaveOccurred())
+		})
+	}
+}
+
+func getClusterScope(client client.Client) (*scope.ClusterScope, error) {
+	cluster := &clusterv1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test",
+		},
+	}
+	cs, err := scope.NewClusterScope(scope.ClusterScopeParams{
+		Client:     client,
+		Cluster:    cluster,
+		AWSCluster: &infrav1.AWSCluster{},
+	})
+	if err != nil {
```

Comment:
To expect error during our tests, we can use gomega.Expect() here

Reply:
Replied in other duplicate comment

Reply:
Replied above